elide-core/src/test/java/com/yahoo/elide/core/UpdateOnCreateTest.java

```diff
@@ -0,0 +1,242 @@
+        dictionary.bindEntity(Book.class);
+        dictionary.bindEntity(UpdateAndCreate.class);
+
+        UpdateAndCreate updateAndCreateNewObject = new UpdateAndCreate();
+        updateAndCreateNewObject.setId(1L);
+        UpdateAndCreate updateAndCreateExistingObject = new UpdateAndCreate();
+        updateAndCreateExistingObject.setId(2L);
+        Book book = new Book();
+        Author author = new Author();
+
+        DataStoreTransaction tx = mock(DataStoreTransaction.class, Answers.CALLS_REAL_METHODS);
+
+        User userOne = new User(1);
+        userOneScope = new RequestScope(null, null, tx, userOne, null, elideSettings, false);
+        User userTwo = new User(2);
+        userTwoScope = new RequestScope(null, null, tx, userTwo, null, elideSettings, false);
+        User userThree = new User(3);
+        userThreeScope = new RequestScope(null, null, tx, userThree, null, elideSettings, false);
+
+        when(tx.createNewObject(UpdateAndCreate.class)).thenReturn(updateAndCreateNewObject);
+        when(tx.loadObject(eq(UpdateAndCreate.class),
+                eq((Serializable) CoerceUtil.coerce(1, Long.class)),
+                eq(Optional.empty()),
+                any(RequestScope.class)
+        )).thenReturn(updateAndCreateExistingObject);
+        when(tx.loadObject(eq(Book.class),
+                eq((Serializable) CoerceUtil.coerce(1, Long.class)),
+                eq(Optional.empty()),
+                any(RequestScope.class)
+        )).thenReturn(book);
+        when(tx.loadObject(eq(Author.class),
+                eq((Serializable) CoerceUtil.coerce(1, Long.class)),
+                eq(Optional.empty()),
+                any(RequestScope.class)
+        )).thenReturn(author);
+    }
+
+    @Test
+    public void testUpdateOnCreateUserOne() {
+
+        PersistentResource<UpdateAndCreate> created = PersistentResource.createObject(null, UpdateAndCreate.class, userOneScope, Optional.of("uuid"));
+        PersistentResource<Book> loadedBook = PersistentResource.loadRecord(Book.class,
+                "1",
+                userOneScope);
+        PersistentResource<Author> loadedAuthor = PersistentResource.loadRecord(Author.class,
+                "1",
+                userOneScope);
+
+        created.updateAttribute("name", "");
+        created.getRequestScope().getPermissionExecutor().executeCommitChecks();
+        Assert.assertThrows(ForbiddenAccessException.class, () -> {
+            created.updateAttribute("type", UpdateAndCreate.AuthorType.CONTRACTED);
+            created.getRequestScope().getPermissionExecutor().executeCommitChecks();
+        });
+        created.updateAttribute("alias", "");
+        created.getRequestScope().getPermissionExecutor().executeCommitChecks();
+        created.addRelation("books", loadedBook);
+        created.getRequestScope().getPermissionExecutor().executeCommitChecks();
+        Assert.assertThrows(ForbiddenAccessException.class, () -> {
```

Comment:
Having multiple assertions in a single test is probably a indication that you should **separate it into multiple tests**, so that one test only fail for one reason.